@jmsperu , unit test error on both deb and rpm:
```
16:19:33 [ERROR]   LibvirtComputingResourceTest.testPlugNicCommandNoMatchMack:3563 ? NullPointer
```

--- a/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtPlugNicCommandWrapper.java
+++ b/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtPlugNicCommandWrapper.java
@@ -122,6 +122,14 @@ public final class LibvirtPlugNicCommandWrapper extends CommandWrapper<PlugNicCo
         try {
             String domXml = vm.getXMLDesc(0);
 
+            // Defensive: getXMLDesc can return null on certain libvirt error paths (and is
+            // null in unit tests where the Domain mock isn't stubbed for this call). Fall
+            // back to letting libvirt auto-assign the PCI slot.
+            if (domXml == null) {
+                logger.debug("Domain XML unavailable, letting libvirt auto-assign PCI slot");
+                return null;
+            }
+
             // Parse all PCI slot numbers currently in use
             Set<Integer> usedSlots = new HashSet<>();
             Pattern slotPattern = Pattern.compile("slot='0x([0-9a-fA-F]+)'");

--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -3548,6 +3548,15 @@ public class LibvirtComputingResourceTest {
             when(vifDriver.plug(nic, "Other PV", "", null)).thenReturn(interfaceDef);
             when(interfaceDef.toString()).thenReturn("Interface");
 
+            // Stub vm.getXMLDesc(0) so findNextAvailablePciSlot can scan the domain XML
+            // for in-use PCI slots. Returning a minimal <domain> with a single NIC at
+            // slot 0x03 exercises the production parser without forcing the production
+            // code into its null-fallback path.
+            when(vm.getXMLDesc(0)).thenReturn(
+                    "<domain><devices><interface type='bridge'>" +
+                    "<address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>" +
+                    "</interface></devices></domain>");
+
             final String interfaceDefStr = interfaceDef.toString();
             doNothing().when(vm).attachDevice(interfaceDefStr);
 

